### PR TITLE
Add JDK 17 to standard builds and update EA builds to JDK 18.

### PR DESCRIPTION
**Pull request by donraab**
Signed-off-by: Donald Raab <Donald.Raab@bnymellon.com>

### CODE REVIEW

**Comment by nikhilnanivadekar** [comment]
We should move checkstyle to 17.

File: .github/workflows/unit-no-p2.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest]
-        java-version: [15, 16]
+        java-version: [15, 16, 17]
```
Comment (nikhilnanivadekar) [inline .github/workflows/unit-no-p2.yml:14]: We should probably only have it as 17

**Comment by nikhilnanivadekar** [comment]
This one should only run on 8 and 11 : https://github.com/eclipse/eclipse-collections/blob/master/.github/workflows/unit.yml

**Review by nikhilnanivadekar — CHANGES_REQUESTED** [review]

**Comment by donraab** [comment]
> We should move checkstyle to 17.

I tried and it failed. I rolled back to 11.

**Comment by donraab** [comment]
> This one should only run on 8 and 11 : https://github.com/eclipse/eclipse-collections/blob/master/.github/workflows/unit.yml

Done.

File: .github/workflows/unit-no-p2.yml (hunk 1 above)
Comment (donraab) [reply .github/workflows/unit-no-p2.yml]: Done

**Comment by donraab** [comment]
Ready to merge.

**Review by nikhilnanivadekar — APPROVED** [review]

**Pull request merged**